### PR TITLE
Validate code snippets in datasets documentation - Part 1

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
## Description
This follows on from #1743. In this PR I have tested _all the datasets that can be instantiated locally_. The following datasets remain to be tested in part 2:

- dask.ParquetDataSet
- holoviews.HoloviewsWriter
- pandas.GBQQueryDataSet
- pandas.GBQTableDataSet
- pandas.SQLQueryDataSet
- pandas.SQLTableDataSet
- redis.PickleDataSet
- all Spark datasets
- tensorflow.TensorFlowModelDataset

The main changes in this PR are as follows:

- Some code blocks had duplicate lines for providing filepath arguments. I believe this was to illustrate both local and external (S3/GCS) filesytems, but after discussion with @MerelTheisenQB we have agreed this is a bit redundant.
- Some comments rendered with weird line breaks. This was fixed.
- There was a comment for a dependency to be installed within an example code block. This has been moved out to `extras_require` in `setup.py`
- Added missing imports

Also warranting a flag 🚩: **GraphMLDataSet and GMLDataSet appear to be identical in purpose and in implementation**

#### Both the YAML and Python examples have been tested under the following setup:

- New conda environment
- `pip install kedro`
- `pip install kedro[all]`

##### For Python:

`kedro jupyter lab`
Code snippets were copied and pasted over
An additional cell was added to check `<dataset name>.exists()` returned True

##### For YAML:

`kedro new -s spaceflights`
The dataset was added to the data catalog in `conf/base/catalog.yml` and added to the pipeline arbitrarily.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1962"></a>

### CODE REVIEW

File: kedro/extras/datasets/json/json_dataset.py [hunk 1]
```diff
@@ -28,17 +28,13 @@ class JSONDataSet(AbstractVersionedDataSet[Any, Any]):
         >>> json_dataset:
         >>>   type: json.JSONDataSet
         >>>   filepath: data/01_raw/location.json
-        >>>   load_args:
```
Comment (AhdraMeraliQB) [inline kedro/extras/datasets/json/json_dataset.py:31]: JSON Dataset does not support the `load_args` parameter: https://kedro.readthedocs.io/en/stable/kedro.extras.datasets.json.JSONDataSet.html#kedro.extras.datasets.json.JSONDataSet.__init__

**Comment by MerelTheisenQB** [comment]
I found the PR where the GraphMLDataSet and GMLDataSet were added: https://github.com/kedro-org/kedro/pull/881 It looks like both of them were added because "Included GraphML in addition to GML. Some tools prefer GraphML as it's able to perfectly preserve internal property maps ([ref](https://graph-tool.skewed.de/static/doc/graph_tool.html#graph_tool.load_graph))." 

So let's just keep them both.

File: kedro/extras/datasets/json/json_dataset.py (hunk 1 above)
Comment (MerelTheisenQB) [reply kedro/extras/datasets/json/json_dataset.py:31]: Great catch!

File: setup.py [hunk 2]
```diff
@@ -78,6 +78,7 @@ def _collect_requirements(requires):
     "pandas.XMLDataSet": [PANDAS, "lxml~=4.6"],
     "pandas.GenericDataSet": [PANDAS],
 }
+pickle_require = {"pickle.PickleDataSet": ["compress-pickle~=2.1.0"]}
```
Comment (MerelTheisenQB) [inline setup.py:81]: The original comment said: "# Add "compress_pickle[lz4]" to requirements.txt" Do we need to specify [lz4] for this requirement?

**Review by MerelTheisenQB — APPROVED** [review]
Great work on going through all these examples and testing them! 👍 ⭐

**Comment by AhdraMeraliQB** [comment]
@MerelTheisenQB 
> I found the PR where the GraphMLDataSet and GMLDataSet were added: #881 It looks like both of them were added because "Included GraphML in addition to GML. Some tools prefer GraphML as it's able to perfectly preserve internal property maps ([ref](https://graph-tool.skewed.de/static/doc/graph_tool.html#graph_tool.load_graph))."
> 
> So let's just keep them both.

I took a look at the ref but couldn't see the preference; is it the name that they prefer? As the implementation is carbon copies 😬

**Review by ankatiyar — APPROVED** [review]
Looks good to me!

**Comment by MerelTheisenQB** [comment]
> I took a look at the ref but couldn't see the preference; is it the name that they prefer? As the implementation is carbon copies 😬

The difference between `GraphML` and `GML` (apparently) is that `GraphML` preserves internal property maps, but `GML` doesn't. See this comment in the docs: "The only file formats which are capable of perfectly preserving the internal property maps are “gt” and “graphml”. Because of this, they should be preferred over the other formats whenever possible." https://graph-tool.skewed.de/static/doc/graph_tool.html#graph_tool.load_graph

Perhaps the question is whether our users make use of both the datasets or not. It's not an extreme maintenance burden so I'd just keep them both.

File: setup.py (hunk 2 above)
Comment (AhdraMeraliQB) [reply setup.py:81]: Good catch - this is what the `pip list` showed after manually installing `compress_pickle[lz4]`, I'll make the change to follow the dask example further up

**Pull request merged**